arwen/host/asyncCall.go

```diff
@@ -122,7 +123,7 @@ func (host *vmHost) sendCallbackToCurrentCaller() error {
 	return nil
 }
 
-func (host *vmHost) sendStorageCallbackToDestination(asyncCallInitiator vmcommon.AsyncInitiator) error {
+func (host *vmHost) sendStorageCallbackToDestination(asyncCallInitiator arwen.AsyncInitiator) error {
```

Comment:
The storage in the name referred to the fact that this function is sending a callback that it could not initiate on the initial contract execution (because it also generated a cross shard async call) an saved the callback information in storage so it can generate it when all it's own callbacks are executed. But I agree, it's a poor choice of name, especially now that after your feedback we no longer have AsyncInitiator and the signature bacame sendStorageCallbackToDestination(callerAddress, returnData []byte).